src/skmultiflow/demos/_test_filters.py

```diff
@@ -1,32 +1,39 @@
-from skmultiflow.data.file_stream import FileStream
-from skmultiflow.transform import MissingValuesCleaner
+import os
+import time
+import pandas as pd
+from streamz import Source
 
+from src.skmultiflow.data import DataGenerator
+from sklearn.impute import SimpleImputer
+
+curr_dir = dir_path = os.path.dirname(os.path.realpath(__file__))
 
 def demo():
     """ _test_filters
-    
-    This demo test the MissingValuesCleaner filter. The transform is set 
-    to clean any value equal to -47, replacing it with the median value 
-    of the last 10 samples, or less if there aren't 10 samples available. 
-    
-    The output will be the 10 instances used in the transform. The first 
-    9 are kept untouched, as they don't have any feature value of -47. The 
-    last samples has its first feature value equal to -47, so it's replaced 
+
+    This demo test compatibility between `sklearn.impute`. The transform is set
+    to clean any value equal to -47, replacing it with the median value
+    of the last 10 samples, or less if there aren't 10 samples available.
+
+    The output will be the 10 instances used in the transform. The first
+    9 are kept untouched, as they don't have any feature value of -47. The
+    last samples has its first feature value equal to -47, so it's replaced
     by the median of the 9 first samples.
-    
-    """
-    stream = FileStream('../data/datasets/covtype.csv', -1, 1)
-    stream.prepare_for_use()
 
-    filter = MissingValuesCleaner(-47, 'median', 10)
+    """
+    df = pd.read_csv(os.path.join(curr_dir, '../data/datasets/covtype.csv'))
+    gen = DataGenerator(df, return_np=True)
+    stream = Source.from_generator(gen, batch_size=1, poll_interval=.001)
+    sliding_array = stream.sliding_window(10).map(pd.np.concatenate)  # get last ten elements
+    data_from_stream = sliding_array.map(SimpleImputer(missing_values=-47, strategy='median').fit_transform).sink_to_list()
+    stream.start()
+    time.sleep(1)  # wait long enough to let all batches pass through the stream
```

Comment:
I would recommend against this in tests, but maybe OK in a demo. There are helpers like [wait_for](https://github.com/python-streamz/streamz/blob/master/streamz/utils_test.py#L107) that could be useful.

Reply:
I did not know there was a function for this 
Thanks !!

Reply:
Actually @martindurant this is not a test, this is a demo.
In the future if demos are replaced by notebooks (I raised an issue for this)
Then calls to sleep or wait_for would not make sense
The cells of the notebook would have to be executed in time, that's all